pkg/sources/github/github.go

```diff
@@ -150,6 +155,62 @@ func (s *Source) Init(aCtx context.Context, name string, jobID, sourceID int64,
 	return nil
 }
 
+func (s *Source) isPublic(repoURL string) bool {
+	public, exists := s.publicMap[repoURL]
+	if exists {
+		return public
+	}
+	log.Debugf("Checking public status for %s", repoURL)
+	if _, unauthenticated := s.conn.GetCredential().(*sourcespb.GitHub_Unauthenticated); unauthenticated {
+		log.Warn("Unauthenticated scans cannot determine if a repository is private.")
+		s.publicMap[repoURL] = false
+		return false
+	}
```

Comment:
Shouldn't we just assume public visibility here?

Reply:
When the repo can't be found, but can be cloned, and the auth type is unauthenticated its probably because the repo is private. I adjusted the logic to reflect that better.